### PR TITLE
[#23] [UI] As a user, I can see the rating view in the Question page

### LINKED ISSUE #23: [UI] As a user, I can see the rating view in the Question page
## Why
The user can see the questions and the corresponding answers in the Question page so he can answer them.

## Acceptance Criteria
- Create the rating component.
- The user can click on an emoji to change the rating.
[Figma](https://www.figma.com/file/i1AI0IzCWuJApX5CK0DmiW/Challenge---Web-App?type=design&node-id=70-370&t=apcSQ1nCik2OLC0u-0)
image

### PR DESCRIPTION
**Pull request by manh-t**
Close #23 

## What happened 👀

Create the Rating component for display type: heart, thumbs, star.

## Insight 📝

N/A

## Proof Of Work 📹

image

image

image

image

### CODE REVIEW

**Comment by manh-t** [comment]
🚀 Deployed on https://64f93a8bd319181ba7dfbd07--manh-react-survey.netlify.app

**Comment by manh-t** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Coverage is a little low, take a look</td>
    </tr>
  </tbody>
</table>

## Coverage in All Files
File | Line Coverage | Statement Coverage | Function Coverage | Branch Coverage
---- | ------------: | -----------------: | ----------------: | --------------:
[src/types/question.ts](../blob/2672a05/src/types/question.ts) | (1/2) 50% | (1/2) 50% | (0/1) 0% | (0/2) 0%
[src/../TextArea/index.tsx](../blob/2672a05/src/components/TextArea/index.tsx) | (5/8) 63% | (5/8) 63% | (1/2) 50% | (0/2) 0%
[src/../AppSlider/index.tsx](../blob/2672a05/src/components/AppSlider/index.tsx) | (4/6) 67% | (4/6) 67% | (1/3) 33% | (3/3) 100%
[src/tests/error.ts](../blob/2672a05/src/tests/error.ts) | (2/3) 67% | (2/3) 67% | (1/2) 50% | (1/3) 33%
[src/../Rating/index.tsx](../blob/2672a05/src/components/Rating/index.tsx) | (18/22) 82% | (18/22) 82% | (7/7) 100% | (4/13) 31%
[src/../Content/index.tsx](../blob/2672a05/src/components/Dashboard/Content/index.tsx) | (14/15) 93% | (14/15) 93% | (7/7) 100% | (6/9) 67%
[src/helpers/deserializer.ts](../blob/2672a05/src/helpers/deserializer.ts) | (17/18) 94% | (17/18) 94% | (7/7) 100% | (10/12) 83%
[src/../v1/requestManager.ts](../blob/2672a05/src/lib/request/v1/requestManager.ts) | (19/20) 95% | (19/20) 95% | (6/6) 100% | (3/5) 60%
[src/App.tsx](../blob/2672a05/src/App.tsx) | (3/3) 100% | (3/3) 100% | (1/1) 100% | (0/0) NaN%
[src/index.tsx](../blob/2672a05/src/index.tsx) | (3/3) 100% | (3/3) 100% | (0/0) NaN% | (0/0) NaN%
[src/../Authentication/index.ts](../blob/2672a05/src/adapters/Authentication/index.ts) | (4/4) 100% | (4/4) 100% | (2/2) 100% | (0/0) NaN%
[src/adapters/Base/index.ts](../blob/2672a05/src/adapters/Base/index.ts) | (8/8) 100% | (8/8) 100% | (2/2) 100% | (2/2) 100%
[src/adapters/BaseAuth/index.ts](../blob/2672a05/src/adapters/BaseAuth/index.ts) | (10/10) 100% | (10/10) 100% | (4/4) 100% | (2/2) 100%
[src/adapters/Survey/index.ts](../blob/2672a05/src/adapters/Survey/index.ts) | (2/2) 100% | (2/2) 100% | (2/2) 100% | (0/0) NaN%
[src/adapters/User/index.ts](../blob/2672a05/src/adapters/User/index.ts) | (1/1) 100% | (1/1) 100% | (1/1) 100% | (0/0) NaN%
Other (38 more) | (247/247) 100% | (247/247) 100% | (88/89) 99% | (76/81) 94%
Total | (358/372) 96% | (358/372) 96% | (130/136) 96% | (107/134) 80%

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against ad3247b84085711d46ea0b167535ef5bc876d9ef
</p>

File: src/types/question.ts [hunk 1]
```diff
@@ -16,3 +16,26 @@ export interface Question extends Resource {
...
+  Outro = 'outro',
+  Unknown = '',
+}
+
+export const getDisplayTypeEnum = (question: Question): DisplayType => {
+  try {
+    return (<never>DisplayType)[question.displayType ?? ''];
+  } catch (error) {
+    return DisplayType.Unknown;
+  }
```
Comment (nvminhtue) [inline src/types/question.ts]: isn't it?
```suggestion
 question.displayType ? DisplayType)[question.displayType] : DisplayType.Unknown;
```

Comment (manh-t) [reply src/types/question.ts]: @nvminhtue thanks, updated it 👍

File: src/components/Rating/index.test.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  describe('given the last value is clicked', () => {
+    it('returns the proper value', () => {
+      let selectedValue = '';
+      const onValueChanged = (answer: Answer) => {
+        selectedValue = answer.id;
+      };
+      const ratingProps = {
+        questionId: 'question-id',
+        items: [
+          { id: '1', resourceType: 'answer', text: 'sdf' },
```
Comment (hoangmirs) [inline src/components/Rating/index.test.tsx]: Could you add some meaningful words instead of `sdf`?

**Review by hoangmirs — APPROVED** [review]

File: src/components/Rating/index.test.tsx (hunk 2 above)
Comment (manh-t) [reply src/components/Rating/index.test.tsx]: @hoangmirs I update to use fabricator for generating test data instead, thanks for reminding 🙏

**Review by nvminhtue — APPROVED** [review]

**Pull request merged**